test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -49,6 +49,22 @@ namespace test
 
 BOOST_FIXTURE_TEST_SUITE(SolidityEndToEndTest, SolidityExecutionFramework)
 
+BOOST_AUTO_TEST_CASE(transaction_status)
+{
+	char const* sourceCode = R"(
+		contract test {
+			function f() { }
+			function g() { revert(); }
+		}
```

Comment:
Please also add `h()` with `assert(false)` (this will test the `invalid` opcode, while `revert()` tests the `revert` opcode) and test for unsuccessful.